v1/plugin/plugin.go

```diff
@@ -356,39 +427,495 @@ type preamble struct {
 	ErrorMessage  string
 }
 
-func startPlugin(srv server, m *meta, p *pluginProxy) int {
-	l, err := net.Listen("tcp", ":"+listenPort)
+func startPlugin(c *cli.Context) error {
+	var (
+		server      *grpc.Server
+		meta        *meta
+		pluginProxy *pluginProxy
+	)
+	libInputOutput.setContext(c)
+	arg, err := processInput(c)
+	if err != nil {
+		return err
+	}
+	if lvl := c.Int("log-level"); lvl > arg.LogLevel {
+		log.SetLevel(log.Level(lvl))
+	} else {
+		log.SetLevel(log.Level(arg.LogLevel))
+	}
+	logger := log.WithFields(log.Fields{
+		"_block": "startPlugin",
+	})
+	switch plugin := appArgs.plugin.(type) {
+	case Collector:
+		proxy := &collectorProxy{
+			plugin:      plugin,
+			pluginProxy: *newPluginProxy(plugin),
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(collectorType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
```

Comment:
Is that panic is intentional or should it be changed to `return err`?

### Current
```
$  ./snap-plugin-collector-rand  --stand-alone --tls

panic: failed to enable TLS for plugin - need both CertPath and KeyPath

goroutine 1 [running]:
panic(0x7cac80, 0xc4201b00e0)
        /root/.gvm/gos/go1.7.3/src/runtime/panic.go:500 +0x1a1
github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin.startPlugin(0xc42001a8c0, 0xc42001a8c0, 0xc420163bef)
        /home/iza/Projects/src/github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/plugin.go:458 +0xe19
github.com/intelsdi-x/snap-plugin-lib-go/vendor/github.com/urfave/cli.HandleAction(0x7bf720, 0x885918, 0xc42001a8c0, 0xc420076300, 0x0)
        /home/iza/Projects/src/github.com/intelsdi-x/snap-plugin-lib-go/vendor/github.com/urfave/cli/app.go:485 +0xd4
github.com/intelsdi-x/snap-plugin-lib-go/vendor/github.com/urfave/cli.(*App).Run(0xc4200929c0, 0xc42000c120, 0x3, 0x3, 0x0, 0x0)
        /home/iza/Projects/src/github.com/intelsdi-x/snap-plugin-lib-go/vendor/github.com/urfave/cli/app.go:259 +0x74f
github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin.StartCollector(0xa3a2a0, 0xa7a128, 0x85425e, 0x13, 0x1, 0x0, 0x0, 0x0, 0xc4200001a0)
        /home/iza/Projects/src/github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/plugin.go:353 +0x17c
main.main()
        /home/iza/Projects/src/github.com/intelsdi-x/snap-plugin-lib-go/examples/snap-plugin-collector-rand/main.go:33 +0x95
```

### How it might look: 
(panic changed to return err)
```
$ ./snap-plugin-collector-rand  --stand-alone --tls
ERRO[0000] failed to enable TLS for plugin - need both CertPath and KeyPath  _block=StartCollector
```